### ISSUE TITLE
4.2 Design Temporal Segmentation Model

### ISSUE THREAD

**Issue by hughmancoder** [hughmancoder/ML-based-analysis-of-sound#5]
- [x] Yamnet
- [ ] RNN model  (use the models in our literature review)

- [ ] Adopt a CRNN-based architecture (Convolutional Recurrent Neural Network) from the following paper and reproduce experiment

[Onset Detection for String Instruments Using Bidirectional Temporal and Convolutional Recurrent Networks](https://dl.acm.org/doi/10.1145/3616195.3616206)

## Training Process

- []  Pre-train the model on QTDS/Böck dataset.
- [ ]  Fine-tune on the curated martial-arts + Jingju dataset (if appropriate)
- [x]  Use Librosa for spectrogram generation. (you can integreate the pipeline built in our classifier model)

## training

- [ ]  Train CRNN for frame-wise onset classification using specifications from: the paper Onset Detection for String Instruments Using Bidirectional Temporal and Convolutional Recurrent Networks

## Model Evaluation

- [x]  Use F1-score as primary evaluation metric.
- [ ]  Apply data split: 70% train / 15% validation / 15% test.
- [ ]  Merge detected onsets within 30 ms into a single event. (choose appropriate threshold)
- [x]  Count an onset/offset as correct if within ±50 ms of ground truth. (choose appropriate threshold)
